src/operations/execute_operation.ts

```diff
@@ -136,11 +136,15 @@ export function executeOperation<
 
     try {
       executeWithServerSelection<TResult>(topology, session, operation, (error, result) => {
+        const asyncResource = operation.options.asyncResource;
```

Comment:
Is there a reason not to create an `asyncResource` for all `AbstractOperation`s? That seems semantically correct and would broaden async_hooks support without introducing more complexity (possibly even removing, as `maybeRunInAsyncScope` becomes unnecessary).

Reply:
That's a valid point. I just assumed there was a reason to bind the resource's lifecycle to the cursor like in the original PR. But on second thought the old codebase didn't use a class for operations so that might be the reason behind the choice. @mbroadst can you please confirm if that is the case and if you see any issues with binding the AsyncResource to operations?
I will do some testing with @addaleax 's version and see if there are any technical issues.

Reply:
That's a valid point. I just assumed it was better to bind the resource's lifecycle to the cursor like in the original PR. But on second thought it might have been because there was no operation class in the old codebase. @mbroadst can you please confirm that and give your thoughts on this? I will test @addaleax 's version as well to check if there are any technical issues with it.